### PR TITLE
[WIP] Create schema for annotation token

### PR DESCRIPTION
**Pull request by nlisgo**

### CODE REVIEW

File: src/api.raml [hunk 1]
```diff
@@ -185,6 +185,28 @@ traits:
...
+            type: string
+    get:
+        description: |
+            Get an annotation token.
+        responses:
+            200:
+                description: |
+                    Annotation token.
+                body:
+                    application/vnd.elife.annotation-token+json;version=1:
```
Comment (giorgiosironi) [inline src/api.raml:205]: I wonder if there is a standard way to mark this as experimental as we will need to merge to go live, and do breaking changes later. Perhaps there is a standard field in RAML? Or could we use `version=0`?

File: src/misc/jwt.v1.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+$schema: http://json-schema.org/draft-04/schema#
+
+title: Token
+type: string
```
Comment (giorgiosironi) [inline src/misc/jwt.v1.yaml]: link to where the regex comes from?
and is it worth it to lock down?

File: src/model/annotation-token.v1.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+$schema: http://json-schema.org/draft-04/schema#
+title: Annotation token
+type: object
+allOf:
+  - properties:
+        token:
+            $ref: ../misc/jwt.v1.yaml
+        expire:
```
Comment (giorgiosironi) [inline src/model/annotation-token.v1.yaml]: we've seen in `profiles` `expires_at` to distuinguish between moments in time and intervals (which would be `expires_in`). Perhaps here too?

Comment (nlisgo) [reply src/model/annotation-token.v1.yaml]: Will change to `expiresAt`

File: src/misc/jwt.v1.yaml (hunk 2 above)
Comment (nlisgo) [reply src/misc/jwt.v1.yaml]: I have found a better regex: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/master/src/System.IdentityModel.Tokens.Jwt/JwtConstants.cs#L58

File: src/api.raml (hunk 1 above)
Comment (nlisgo) [reply src/api.raml:205]: Let's discuss this with @thewilkybarkid.

Comment (thewilkybarkid) [reply src/api.raml:205]: RAML has annotations. Thinking about it, we should do that for deprecations too. I'll take a look at adding them.

File: src/samples/annotation-token/v1/complete.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+    "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhdWQiOiJoeXBvdGhlcy5pcyIsImlzcyI6ImNsaWVudF9pZCIsInN1YiI6ImFjY3Q6dXNlckBhdXRob3JpdHkiLCJuYmYiOjE1MDU0OTU4NzgsImV4cCI6MTUwNTQ5NjQ3OH0.FttNDnvdXMeSVaKVDzoYUW7RsP7ovMVwNmNRVZW8OsE",
+    "expire": "2017-09-15T19:27:58Z"
```
Comment (thewilkybarkid) [inline src/samples/annotation-token/v1/complete.json:3]: `expiresAt`?

File: src/api.raml (hunk 1 above)
Comment (thewilkybarkid) [reply src/api.raml:205]: We've started using them, so the whole API should be `@experimental`.

**Comment by thewilkybarkid** [comment]
This isn't needed.

**Pull request closed without merging**